test/gtest/common/test_helpers.cc

```diff
@@ -504,10 +504,18 @@ static std::set<std::string> get_all_rdmacm_net_devices()
      * device.
      */
     for (size_t i = 0; i < ndevs.size(); ++i) {
-        std::string infiniband_dir = sysfs_net_dir + "/" + ndevs[i] +
-                                     "/device/infiniband";
-        if (!read_dir(infiniband_dir).empty()) {
-            devices.insert(ndevs[i]);
+        std::string infiniband_dir          = sysfs_net_dir + "/" + ndevs[i] +
+                                              "/device/infiniband";
+        std::vector<std::string> ib_devices = read_dir(infiniband_dir);
+
+        for (auto &ib_device : ib_devices) {
+            std::string ports_dir             = infiniband_dir + "/" +
+                                                ib_device + "/ports";
+            std::vector<std::string> ib_ports = read_dir(ports_dir);
+
+            for (auto &ib_port : ib_ports) {
+                devices.emplace(ndevs[i], ib_device + ":" + ib_port);
```

Comment:
seems only one ib dev will be saved eventually per ndev, so why need to loop over all devices and ports?

Reply:
> seems only one ib dev will be saved eventually per ndev, so why need to loop over all devices and ports?

yes, it should be only one `device:port` pair.
maybe better to add assertion - to  make sure it is true?

Reply:
IMO no need, just take first one as previously

Reply:
ok, done